### ISSUE TITLE
Discrepancy between reward model training tokenization and PPO tokenization in PPOTrainerv2

### ISSUE THREAD

**Issue by linyuhongg** [huggingface/trl#1702]
During reward model training, the RM is trained on tokenize(input + response). But, during PPO, when calling the RM, it does torch.cat(tokenize(input) + tokenize(response)) (cf https://github.com/huggingface/trl/blob/275d33b3ef4f7afd40f79cc53591659bacfa3499/trl/trainer/ppov2_trainer.py#L556). 

Does this discrepancy in how data is processed during PPO training for the reward model affect performance?

**Comment by vwxyzjn** [comment]
You got to be very careful with prompts.

> the RM is trained on tokenize(input + response).

For example, let's say we are working with a chat-based task, the prompt is `xxxx, Assistant:`, then the response should be  ` happy to help<endoftext>`. Note that there is the leading space in the response.

During PPO training, if you maintain the same prompt, then things should work as expected.

**Comment by linyuhongg** [comment]
I see, thank you for the quick response.
